### PR TITLE
[OpenMP][flang] Adding more tests for commonblock with target map

### PR DESCRIPTION
**Pull request by shraiysh**
This patch addresses the concern about multiple devices and also adds more tests for `map(to:)`, `map(from:)` and named common blocks.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
@agozillon @TIFitis Are you OK with these tests?

File: openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90 [hunk 1]
```diff
@@ -80,3 +87,61 @@ subroutine commonblock_simple_with_real
...
+  print *, "var4 after target = ", var4
+  print *, "tmp after target = ", tmp
+end subroutine
+
+! CHECK: var4 after target = 10
+! CHECK: tmp after target = 10
+
+! -----
+
+subroutine commonblock_simple_from
```
Comment (TIFitis) [inline openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90:112]: Test `commonblock_simple_to` is already testing the `for` map-type isn't it?

File: openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90 [hunk 2]
```diff
@@ -26,11 +30,14 @@ subroutine check_device
...
+  !CHECK: [[DD:[0-9]+]]
+  !CHECK: devices: [[ND]] [[DD]]
   print *, "devices: ", devices
 end subroutine check_device
 
-!CHECK: devices: 1 0
-
 !-----
 
 subroutine commonblock_simple_with_implicit_type
```
Comment (TIFitis) [inline openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90:43]: Can you remove the explicit mapping from this test? I guess we want to check implicit mapping here.

**Review by TIFitis — APPROVED** [review]
LGTM with the suggested changes.

**Review by agozillon — APPROVED** [review]
LGTM

File: openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90 (hunk 1 above)
Comment (shraiysh) [reply openmp/libomptarget/test/offloading/fortran/target_map_common_block.f90:112]: Yes, that is correct. Removed this redundant test. Thanks!

**Pull request merged**